assets/js/components/devices/DeviceFlows.jsx

```diff
@@ -0,0 +1,193 @@
+import React, { Component } from 'react'
+import { ReactFlowProvider } from 'react-flow-renderer';
+import withGql from '../../graphql/withGql'
+import { connect } from 'react-redux'
+import { Card, Typography } from 'antd';
+import { SkeletonLayout } from '../common/SkeletonLayout';
+const { Text } = Typography
+import { FLOWS_BY_DEVICE } from '../../graphql/flows'
+import FlowsLayout from '../common/FlowsLayout';
+
+class DeviceFlows extends Component {
+refetchEntries = () => {
+  const { refetch } = this.props.deviceFlowsQuery;
+  refetch();
+}
+
+componentDidMount() {
+    const { currentOrganizationId, socket } = this.props;
+
+    this.channel = socket.channel("graphql:flows_update", {});
+    this.channel.join();
+    this.channel.on(`graphql:flows_update:${currentOrganizationId}:organization_flows_update`, (message) => {
+      this.refetchEntries();
+    })
+  }
+
+  componentWillUnmount() {
+    this.channel.leave();
+  }
+
+  addCopyIfNodeExisting = (nodeId, existingElements) => {
+    if (nodeId in existingElements) {
+      let i = 1;
+      while (existingElements[nodeId + "_copy" + i]) {
+        i++;
+      }
+      return `${nodeId}_copy${i}`;
+    } else {
+      return nodeId;
+    }
+  }
+
+  generateElements = (flows) => {
```

Comment:
to clarify, all we are doing here is creating a row for each flow row that is returned from the db right? and making copies of nodes as needed

Reply:
correcto!